opt_einsum/paths.py

```diff
@@ -484,5 +497,11 @@ def eager(inputs, output, idx_dict):
     >>> eager(isets, oset, idx_sizes)
     [(0, 2), (0, 1)]
     """
+    if memory_limit not in _UNLIMITED_MEM:
+        return branch(inputs, output, idx_dict, memory_limit, nbranch=1)
+
     ssa_path = _ssa_optimize(inputs, output, idx_dict)
     return ssa_to_linear(ssa_path)
+
+
+greedy = eager
```

Comment:
I was just trying to maintain maximum backward compatibility. But yes probably cleaner to just rename ``eager->greedy``, unless @fritzo has any objections? ``'eager'`` will still be used as an alias with ``optimize='eager'`` but otherwise will not be referenced.